src/main/java/org/truffleruby/parser/BodyTranslator.java

```diff
@@ -3340,6 +3332,73 @@ private RubyNode addNewlineIfNeeded(ParseNode jrubyNode, RubyNode node) {
         return node;
     }
 
+    private interface ValueFromNode {
```

Comment:
Given this file is already huge, could you move this interface and the two classes in a separate file?
I'd nest the two classes under `ValueFromNode` to keep them together.

Reply:
Can do.

Reply:
Fixed.